### PR TITLE
[silicon_creator] move pinmux attr spin wait time to a macro

### PR DESCRIPTION
**Pull request by timothytrippel**
This addresses a review comment in #17959.

### CODE REVIEW

File: sw/device/silicon_creator/lib/base/chip.h [hunk 1]
```diff
@@ -43,6 +43,15 @@
...
 #define TEST_ROM_IDENTIFIER 0x54534554
 
+/**
+ * Pinmux pull up/down wait delay.
+ *
+ * After enabling the pull-up/down on a pin, we need to wait for ~5us for the
+ * configuration to propagate to the physical pads. 5us is 500 clock cycles
+ * assuming a 100MHz clock.
+ */
+#define PINMUX_PAD_ATTR_PROP_DELAY_CYCS 500
```
Comment (alphan) [inline sw/device/silicon_creator/lib/base/chip.h:53]: nit: for the sake of clarity
```suggestion
#define PINMUX_PAD_ATTR_PROP_CYCLES 500
```

**Review by alphan — APPROVED** [review]

File: sw/device/silicon_creator/lib/base/chip.h (hunk 1 above)
Comment (timothytrippel) [reply sw/device/silicon_creator/lib/base/chip.h]: fixed.

File: sw/device/silicon_creator/lib/drivers/pinmux_unittest.cc [hunk 2]
```diff
@@ -71,6 +73,13 @@ class InitTest : public PinmuxTest {
   };
 };
 
+TEST_F(InitTest, PadAttrPropagationDelay) {
+  const uint64_t kCpuClockPeriodNs =
+      (1.0 / static_cast<double>(kClockFreqCpuHz)) * 1000000000;
```
Comment (alphan) [inline sw/device/silicon_creator/lib/drivers/pinmux_unittest.cc:78]: nit: you don't really need the cast to double and the division if the result will be stored in an integer - please note the digit separator `'` which was added in c++14:
```suggestion
  const uint64_t kCpuClockPeriodNs = 1'000'000'000 / kClockFreqCpuHz;
```

**Review by alphan — APPROVED** [review]
Thanks @timothytrippel !

File: sw/device/silicon_creator/lib/drivers/pinmux_unittest.cc (hunk 2 above)
Comment (dmcardle) [reply sw/device/silicon_creator/lib/drivers/pinmux_unittest.cc:78]: Big +1 to eliminating floating point whenever possible. I've been burned so many times!

Comment (timothytrippel) [reply sw/device/silicon_creator/lib/drivers/pinmux_unittest.cc]: fixed.

**Pull request merged**